### PR TITLE
feat: gateway for all instances

### LINKED ISSUE #914: Reuse of components from different instances in automatic tests
**Is your feature request related to a problem? Please describe.**
To increase productivity and reduce maintenance we want to reuse as much components as possible for the different instances of the Devhub portal code. Today code is duplicated across instances, and a fix or new feature might have to be applied several places. One blocker for being able to reuse across instances is that `bos-loader` only serves one instance at a time. We cannot for example serve components that are specific to infrastructure-committee and reference components from devhub when using bos-loader in the test environment.

**Describe the solution you'd like**
We would like to switch to using bos workspace in the test environment, since it has the capability of serving components from all instances at the same time.

**Describe alternatives you've considered**
We have also considered using mulitple instances of bos-loader, or fixing bos-loader, but found that bos-workspace is the closest to be able to fulfill this need. Also it is possible already to use bos-workspace for local development

### PR DESCRIPTION
**Pull request by petersalomonsen**
This PR switch to using [BOS-workspace](https://github.com/NEARBuilders/bos-workspace) as a gateway for tests, because of it's capability to serve multiple instances at the same time from the local components. Also BOS-workspace has built in the functionality for not having to use a redirect-map, which means that the RPC proxy developed for this particular purpose can be dropped from this project. This reduce complexity in the test code. The flags setting for localstorage is removed, and also the mocking of RPC requests for getting components from bos-loader which was used in some tests.

One feature that we have in devhub, but not in bos-workspace is the caching of RPC requests. To compensate for the lack of this, timeouts are increased. This is more realistic for users that have never visited devhub before, but probably not realistic when it comes to returning users since they will have a much faster loading experience than what we see in the tests. Also dropping this cache might re-introduce some more flakyness in the tests. However, removing the cache revealed more clearly some specific flakyness issues, which have been adressed in this PR. Issues such as with VM.require as mentioned below and waiting for labels to appear in the sunset tests. If re-introducing cache in the future it might also be worth considering using client side indexeddb, which is the same as the NearSocialVM use, and not at the RPC level.

Note that bos workspace is used with the `-g` parameter, to specify a custom gateway. There is the script [scripts/create-static-hosting-web-gateway-folder.mjs](scripts/create-static-hosting-web-gateway-folder.mjs) to create a temporary folder with the html/js files to be served from the gateway, using [NEAR-bos-webcomponent](https://github.com/NEARBuilders/near-bos-webcomponent).

There is also a change in the code. The github addon configurator test was suddenly struggling when not having an RPC cache, and this can be seen in the live instance also. There is no "Board" section, which is again because the board state columns are not populated when having to wait for required widgets to load ( VM.require ). Functions loaded with VM.require should not be set to empty functions while waiting for load, but rather the component should return and show a loading indicator. This is especially the case when using VM.require functions to set initial state.

fixes #914

### CODE REVIEW

**Comment by petersalomonsen** [comment]
@race-of-sloths Please include this PR in the race

**Comment by race-of-sloths** [comment]
@petersalomonsen Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/petersalomonsen?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/petersalomonsen)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/petersalomonsen)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/devhub.near/widget/devhub/entity/addon/github/Configurator.jsx
Comment (Megha-Dev-19) [inline instances/devhub.near/widget/devhub/entity/addon/github/Configurator.jsx:104]: we can remove the log?

**Review by Megha-Dev-19 — APPROVED** [review]
Looks great

**Comment by Megha-Dev-19** [comment]
@race-of-sloths score 5

File: instances/devhub.near/widget/devhub/entity/addon/github/Configurator.jsx
Comment (petersalomonsen) [reply instances/devhub.near/widget/devhub/entity/addon/github/Configurator.jsx]: Yes. Thanks for seeing that.

**Review by Tguntenaar — COMMENTED** [review]
Nice, I was already wondering how that would work with the test environments.

**Pull request merged**
